src/massDistribution/Ferriere.cpp

```diff
@@ -0,0 +1,330 @@
+	double R = sqrt(position.x*position.x+position.y*position.y);
+	bool inner = R< 3*kpc;
+		
+	if(inner == true){   //inner
+	
+	double x = position.x/pc;
+	double y = position.y/pc;
+	double z = position.z/pc;
+	
+	//warm interstellar matter
+	double nWIM = 0;		
+	
+	double H = (x*x + pow(y+10.,2))/(145*145);
+	nWIM += exp(-H)* exp(-pow(z+20.,2)/(26*26));
+	nWIM += 0.009*exp(-pow((R/pc-3700)/(0.5*3700),2))*1/pow(cosh(z/140.),2);
+	nWIM += 0.005*cos(M_PI*R/pc*0.5/17000)*1/pow(cosh(z/950.),2);
+	double n3 = nWIM;
+	
+	nWIM *= 8.0/ccm;	//rescaling with density at center
+	
+	
+	//very hot interstellar matter
+	double nVHIM = 0;
+	double alphaVH = 21./180*M_PI;		//angle for very hot IM in radiant 
+	double cosA = cos(alphaVH);
+	double sinA = sin(alphaVH);
+	double etta = y*cosA+z*sinA;		// coordinate transformation for VHIM along major axis
+	double chi = -y*sinA+z*cosA;
+	
+	nVHIM = 0.29/ccm*exp(-((x*x+etta*etta)/(162.*162.)+chi*chi/(90*90)));
+	
+	n = nWIM + nVHIM;
+
+	}
+	else {		// outer region
+		
+		double z = position.z;
+		
+		double nWarm = 0;
+		nWarm += 0.0237/ccm*exp(-(pow(R,2)-pow(Rsun,2))/pow(37*kpc,2))*exp(-fabs(z)/(1*kpc));
+		nWarm += 0.0013/ccm*exp(-(pow(R-4*kpc,2)-pow(Rsun-4*kpc,2))/pow(2*kpc,2))*exp(-fabs(z)/(150*pc));
+		
+		double nHot = 0;
+		nHot += 0.12*exp(-(R-Rsun)/(4.9*kpc));
+		nHot += 0.88*exp(-(pow(R-4.5*kpc,2)-pow(Rsun-4.5*kpc,2))/pow(2.9*kpc,2));
+		nHot *= pow(R/Rsun, -1.65);
+		nHot *= exp(-fabs(z)/((1500*pc)*pow(R/Rsun,1.65)));
+		nHot *= 4.8e-4/ccm;
+		
+		n= nWarm + nHot;
+		
+	}
+		
+	return n;
+}
+
+
+double Ferriere::getH2Density(const Vector3d &position) const{
+
+	double n=0;
+	double R=sqrt(pow(position.x,2)+pow(position.y,2));
+	bool inner = R<3*kpc;	
+	
+	if(inner == true) {		
+	
+		double nCMZ = 0;
+		double nDISK = 0;
+		
+		Vector3d pos =CMZTrafo(position); // coord trafo for CMZ
+		double x = pos.x/pc;	// all units in pc
+		double y = pos.y/pc;
+		double z = pos.z/pc;
+		
+		double A = sqrt(pow(x,2)+pow(2.5*y,2));	// ellipticity
+		nCMZ = exp(-pow((A-125.)/137.,4))*exp(-pow(z/18.,2));
+		nCMZ *= 150/ccm;		// rescaling
+		
+		pos = DISKTrafo(position);	// coord trafo for disk
+		x=pos.x/pc;
+		y=pos.y/pc;
+		z=pos.z/pc;
+		
+		A = sqrt(pow(x,2)+pow(3.1*y,2));
+		nDISK = exp(-pow((A-1200)/438,4))*exp(-pow(z/42,2));
+		nDISK *= 4.8/ccm;		//rescaling
+		
+		n = nCMZ + nDISK;
+	}	
+	else {		//outer region
+		double z = position.z/pc;
+		n = pow(R/Rsun, -0.58);
+		n *= exp(-(pow(R-4.5*kpc,2)-pow(Rsun-4.5*kpc,2))/pow(2.9*kpc,2));
+		n *= exp(-pow(z/(81*pow(R/Rsun,0.58)),2));
+		n *= 0.58/ccm;		//rescaling
+	}
+	
+	return n;
+}
+
+double Ferriere::getDensity(const Vector3d &position) const{ 
+
+	double n=0; 
+	if(isforHI){
+		n += getHIDensity(position);
+	}
+	if(isforHII){
+		n+=getHIIDensity(position);
+	}
+	if(isforH2){
+		n+=getH2Density(position);
+	}
+	
+	//check if any density is activ and give warning if not
+	bool anyDensityActive = isforHI||isforHII||isforH2;
+
+	if(anyDensityActive == false){
```

Comment:
avoid declaring variable if not needed outside, i.e. write directly if (isforH1 | isforHII | ... }.

However, is the warning here necessary? This introduces a check (and potential output) on every get which can be a a lot I guess.